src/lib/storage/value_segment.cpp

```diff
@@ -14,32 +14,29 @@ namespace opossum {
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _values.at(chunk_offset);
```

Comment:
The use of `.at` means that the `ValueSegment`'s `[]` operator throws an error for out-of-bounds `chunk_offset`s. This is different from the `[]` operator's behavior e.g. for vectors, where `[]` is not bounds-checking. I would argue that the `ValueSegment`'s `[]` operator should not be bounds-checking to be consistent with the `[]` behaviour for other common container types.

```suggestion
  return _values[chunk_offset];
```

Reply:
Bound checking makes sense often times. But your point is totally tight in this case. Thanks!